src/hocon.erl

```diff
@@ -434,6 +466,21 @@ nested_put([Key], Val, Map) ->
 nested_put([Key|Paths], Val, Map) ->
     merge(Key, nested_put(Paths, Val, #{}), Map).
 
+nested_update_with(Fun, Map) ->
+    nested_update_with(Fun, Map, maps:iterator(Map)).
+
+nested_update_with(Fun, Map, I) ->
+    case maps:next(I) of
+        {Key, ChildMap, Next} when is_map(ChildMap) ->
+            Updated = #{Key => nested_update_with(Fun, ChildMap)},
+            nested_update_with(Fun, do_deep_merge(Map, Updated), Next);
```

Comment:
why deep merge? key is unique and value is already updated recursively

Reply:
true